Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -31,6 +36,23 @@ extension Style {
         }
     }
 
+    private func isServerSideLocalizationSet() -> Bool {
+        let settings = SettingsServiceFactory.getInstance(storageType: .nonPersistent)
```

Comment:
This check doesn't work when mixing settings storage types. i.e. when language/worldview is set to persistent storage on the app side. In this case this check returns `false`. I wonder if we need to check both storage types here 🤔 
@alexshalamov any thoughts?